src/lib.rs

```diff
@@ -124,171 +124,401 @@ use proc_macro::TokenStream;
-                            out.visibility.span(),
-                            "`pub` visibility breaks the seal as allows to use \
+	fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
+		let mut out = Self::default();
+
+		while !input.is_empty() {
+			let ident = syn::Ident::parse_any(&input.fork())?;
+
+			match ident.to_string().as_str() {
+				"erase" => {
+					syn::Ident::parse_any(input)?;
+					out.erased = true;
+				}
+
+				"pub" => {
+					out.visibility = input.parse()?;
+					if matches!(out.visibility, syn::Visibility::Public(_)) {
+						return Err(syn::Error::new(
+							out.visibility.span(),
+							"`pub` visibility breaks the seal as allows to use \
                              it outside its crate.\n\
                              Consider tightening the visibility (e.g. \
                              `pub(crate)`) if you actually need sealing.",
-                        ));
-                    }
-                }
-
-                unknown => {
-                    return Err(syn::Error::new(
-                        ident.span(),
-                        format!("unknown `{}` attribute argument", unknown),
-                    ))
-                }
-            }
-
-            if input
-                .lookahead1()
-                .peek(token::Comma)
-                .then(|| input.parse::<token::Comma>())
-                .transpose()?
-                .is_none()
-                && !input.is_empty()
-            {
-                return Err(syn::Error::new(ident.span(), "expected followed by `,`"));
-            }
-        }
-
-        Ok(out)
-    }
+						));
+					}
+				}
+
+				"partial" => {
+					syn::Ident::parse_any(input)?;
+					out.partial = true;
+				}
+
+				unknown => {
+					return Err(syn::Error::new(
+						ident.span(),
+						format!("unknown `{}` attribute argument", unknown),
+					))
+				}
+			}
+
+			if input
+				.lookahead1()
+				.peek(token::Comma)
+				.then(|| input.parse::<token::Comma>())
+				.transpose()?
+				.is_none() && !input.is_empty()
+			{
+				return Err(syn::Error::new(ident.span(), "expected followed by `,`"));
+			}
+		}
+
+		Ok(out)
+	}
+}
+
+/// Arguments accepted by `#[seal]` attribute when placed on functions in a (partially) sealed trait
+struct FunctionArguments {
+	/// Determines whether the sealed function is wrapped by a public function that is callable
+	callable: bool,
+}
+
+impl Default for FunctionArguments {
+	fn default() -> Self {
+		FunctionArguments { callable: false }
+	}
+}
+
+impl Parse for FunctionArguments {
+	fn parse(input: ParseStream) -> syn::Result<Self> {
+		let mut out = FunctionArguments::default();
+		let ident = syn::Ident::parse(&input.fork())?;
+
+		match ident.to_string().as_str() {
+			"callable" => {
+				syn::Ident::parse(input)?;
+				out.callable = true
+			}
+			unknown => {
+				return Err(syn::Error::new(
+					ident.span(),
+					format!("unknown `{}` attribute argument", unknown),
```

Comment:
Given we're going to force users to pick between `callable` and `uncallable`, explain which options are available. Something among the lines of:
```suggestion
					format!("unknown `{}` attribute argument. #[seal] only supports `callable` and `uncallable`", unknown),
```